learning_resources/factories.py

```diff
@@ -123,7 +123,7 @@ class Meta:
 class LearningResourceOfferorFactory(DjangoModelFactory):
     """Factory for LearningResourceOfferor"""
 
-    name = FuzzyChoice([offeror.value for offeror in constants.OfferedBy])
+    name = factory.LazyAttribute(lambda obj: constants.OfferedBy[obj.code].value)
     code = FuzzyChoice([offeror.name for offeror in constants.OfferedBy])
     professional = Faker("boolean")
```

Comment:
@mbertrand below is set
```
 django_get_or_create = ("name",)
```
which I think is because `LearningResourceOfferor.name` is unique. Should `code` also be unique, and should we change `django_get_or_create` to use `code` ?

Reply:
I'm pretty certain the actual cause of the errors was that `filters_test.mock_courses` essentially ran
```
LearningResourceOfferor.create(code="ocw") # used a random name
LearningResourceOfferor.create(code="mitx") # used a random name
```
and if the name happened to already exist in the test database, then "code" would be ignored, because 
> When django_get_or_create is used, be aware that any new values passed to the Factory are not used to update an existing model.
>
> [factoryboy docs](https://factoryboy.readthedocs.io/en/stable/orms.html#factory.django.DjangoOptions.django_get_or_create)

Reply:
@ChristopherChudzicki there are some changes related to this in [PR 173](https://github.com/mitodl/mit-open/pull/173/files#diff-4d1ca0172a3b2e3be122db95147c48221f1027a510850fdcc6018a89d894fd9bR125-L128)

Reply:
@mbertrand Ah, cool. Happy to leave it to that for fixing.

What do you think about `code` being unique?

Reply:
Sounds good to me